### PR TITLE
Update link and port in ClickHouse documentation

### PR DESCRIPTION
**Pull request by tylerhannan**
## Description

some extremely minor changes to update the URLs for the website, point to the JDBC driver repository for deeper compatibility information, and resolve an incorrect port in the TLS URL

> Is this change a fix, improvement, new feature, refactoring, or other?
Minor docs update

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)
n/a

> How would you describe this change to a non-technical end user or system administrator?
documentation update

## Documentation

(x) Sufficient documentation is included in this PR.

## Release notes

(x) No release notes entries required.

### CODE REVIEW

**Comment by tylerhannan** [comment]
CLA submitted via email

File: docs/src/main/sphinx/connector/clickhouse.rst [hunk 1]
```diff
@@ -44,8 +44,7 @@ appropriate for your setup:
     by default, but the new driver only supports ClickHouse server with version >= 20.7.
 
     For compatibility with ClickHouse server versions < 20.7,
-    you can temporarily continue to use the old ClickHouse driver(``ru.yandex.clickhouse.ClickHouseDriver``)
-    by adding the following catalog property: ``clickhouse.legacy-driver=true``.
+    please peruse the compatibility information at [ClickHouse JDBC Driver]<https://github.com/ClickHouse/clickhouse-jdbc>
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/clickhouse.rst:47]: Change "peruse" to "see" and the link format is wrong as sphinx documentation.

File: docs/src/main/sphinx/connector/clickhouse.rst [hunk 2]
```diff
@@ -44,8 +44,7 @@ appropriate for your setup:
     by default, but the new driver only supports ClickHouse server with version >= 20.7.
 
     For compatibility with ClickHouse server versions < 20.7,
-    you can temporarily continue to use the old ClickHouse driver(``ru.yandex.clickhouse.ClickHouseDriver``)
-    by adding the following catalog property: ``clickhouse.legacy-driver=true``.
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/clickhouse.rst:48]: Please don't remove this config property.

**Comment by ebyhr** [comment]
Thanks for your contribution! Please follow the commit message guideline: https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#format-git-commit-messages

**Review by ebyhr — APPROVED** [review]

**Comment by ebyhr** [comment]
@tylerhannan I reverted some change about replacement of driver package because we still need to document it. Let me include only link and port change.

**Comment by ebyhr** [comment]
Merged, thanks!

**Pull request merged**